src/spotify.ts

```diff
@@ -13,9 +13,11 @@ export default class Spotify {
 
     }
 
-    async getPlaylist(id: string) {
-        const result = await this.client.getPlaylistTracks(id)
+    async getPlaylist(id: string, offset: number) {
+        const result = await this.client.getPlaylistTracks(id, {
+            offset: offset,
+        });
 
         return result.body.items.map(({ track }) => track)
     }
```

Comment:
I would suggest putting the code to fetch all tracks in this method directly. I believe you want need to make any changes in `music-quiz.ts` by doing this

```suggestion
    async getPlaylist(id: string) {
        let finished = false,
             offset = 0,
             tracks = []

        while (!finished) {
            const result = await this.client.getPlaylistTracks(id, { offset, limit: 100 })
            tracks = tracks.concat(result.body.items.map(({ track }) => track))

            offset += 100
            finished = tracks.length >= result.body.total
        }

        return tracks
    }
```

Reply:
will see what i can do about this in the coming days!